packages/core/src/createGetBalances.ts

```diff
@@ -44,22 +46,40 @@ export const createGetBalances = ({ send }: Provider) =>
      * - `INVALID_THRESHOLD`: Invalid `threshold`
      * - Fetch error
      */
-    (addresses: ReadonlyArray<Hash>, threshold: number, callback?: Callback<Balances>): Promise<Balances> =>
-        Promise.resolve(
+    (
+        addresses: ReadonlyArray<Hash>,
+        threshold: number,
+        tips?: Hash[],
+        callback?: Callback<Balances>
+    ): Promise<Balances> => {
+        // If no tips are provided, switch arguments
+        if (tips && typeof tips === 'function') {
+            callback = tips
+            tips = []
+        }
+
+        return Promise.resolve(
             validate(
-                arrayValidator(hashValidator)(addresses, errors.INVALID_ADDRESS),
-                getBalancesThresholdValidator(threshold)
+                ...[
+                    ...[
+                        arrayValidator(hashValidator)(addresses, errors.INVALID_ADDRESS),
+                        getBalancesThresholdValidator(threshold),
+                    ],
+                    ...(tips && tips.length ? [arrayValidator(transactionHashValidator)(tips)] : []),
```

Comment:
one could pass an object without `length` property as tips, which should be validated I think.

Reply:
Perhaps just best solution is to wrap the validator in another function like this:
```
optional(arrayValidator)
```
We can leave this as is right now and fix it in other PR.

Reply:
Good catch. Have [fixed](https://github.com/iotaledger/iota.js/pull/317/commits/ef10d616c0dd1a9cc3a7438fb0348e8f7dbb9eea) this by checking if provided tips is of type `Array`.